xla/service/gpu/backend_configs.proto

```diff
@@ -197,3 +201,32 @@ message CudnnfMHABackendConfig {
   // Is causal mask
   bool is_causal_mask = 21;
 }
+
+// Generic backend config for XLA:GPU
+message GpuBackendConfig {
+  // Specifies which operation queue the current instruction will run on.
+  // A backend may have multiple operation queues to run instructions
+  // concurrently, use this to signal the backend which queue to dispatch to.
+  // The backend should keep a mapping of
+  // operation_queue_id->actual_hardware_queue_id if runtime will create
+  // different IDs.
+  int64 operation_queue_id = 1;
+
+  // Specifies which operation queues to await for data when running with
+  // multiple operation queues.
+  repeated int64 wait_on_operation_queues = 2;
+
+  CudnnConvBackendConfig cudnn_conv_backend_config = 3;
+
+  GemmBackendConfig gemm_backend_config = 4;
+
+  BitcastBackendConfig bitcast_backend_config = 5;
+
+  CollectiveBackendConfig collective_backend_config = 6;
+
+  FusionBackendConfig fusion_backend_config = 7;
+
+  CudnnNormBackendConfig cudnn_norm_backend_config = 8;
+
+  CudnnfMHABackendConfig cudnn_fmha_backend_config = 9;
```

Comment:
nit: would it make sense to put all these into a `oneof`? Assuming these are already mutually exclusive in the existing code, but I'm not sure if they will ever coexist, @cheshire would be the better person to make the call.

Reply:
i think it makes sense to make it a oneof, they were meant to be individual configs for a given library or a codegen. i changed it to be oneof.